src/pool/srv_pool.c

```diff
@@ -5368,6 +5438,13 @@ ds_pool_update_handler(crt_rpc_t *rpc)
 
 	list.pta_number = in->pti_addr_list.ca_count;
 	list.pta_addrs = in->pti_addr_list.ca_arrays;
+
+	if (opc_get(rpc->cr_opc) == POOL_REINT) {
+		rc = pool_discard(rpc->cr_ctx, svc, &list);
```

Comment:
I have sometimes passed wrong ranks to "dmg reint" commands in my testing. What will happen if this kind of mistakes are made on a production system? For instance, an administrator specifies an existing UP rank by mistake.

Reply:
ah, right, It should validate the ranks before discard indeed. thanks.